tensorflow/compiler/mlir/BUILD

```diff
@@ -51,6 +51,7 @@ cc_library(
         "//tensorflow/compiler/mlir/lite:tensorflow_lite",
         "//tensorflow/compiler/mlir/lite:tf_tfl_passes",  # buildcleaner:keep
         "//tensorflow/compiler/mlir/quantization/stablehlo:bridge_passes",
+        "//tensorflow/compiler/mlir/tensor:tfl_passes",
```

Comment:
Rather keep these inside TOSA directory: mlir folder is a bit of a mix bag and whole different set of testing/groups than is obvious from here. Given the nature of being complementary to TOSA passes for Tflite conversion, it's better to house it there.

Reply:
There are several reasons why I think it might be preferable to actually stick to this file structure:

- The ultimate goal of this pass is being self-contained and independent from existing TOSA lowering (see description of PR). This allows us to reason about its implementation in an independent manner from TOSA semantic restrictions and active evolution. The implementation of the proposed lowering for `tfl.reshape` is an example of a self-contained lowering independent from the existing TFL-to-TOSA conversion. It's a good example of a conversion that is already complex enough as is - let alone if we had to reason about TOSA capabilities on top of it.

- We at MathWorks would like to focus our efforts in maximizing TFL lowering support. Our impression so far is that the TFL-to-TOSA pass has gotten in the way of this, given the philosophy in place of not introducing ops in other dialects. Having the TFL-to-tensor pass in a separate directory would allow us to more freely and quickly upstream additional changes moving forward with less chances for conflicts. We're basically making it explicit in the file system that this development is fully independent from TOSA. It'd be nice if we could "own" this part of the file system for a while as we add new features.

- This pass promises to grow in complexity quickly. Look at the number of functions involved in just lowering `tfl.reshape`! Keeping this in a separate directory will give us freedom to split files as the grow while not polluting the TOSA directory. Compilation time of large files in TOSA is currently too high, negatively affecting feedback cycles when LSPs can't detect errors in complex template uses (e.g., `builder.create` calls).